### PR TITLE
Test SolidusStarterFrontend against Solidus master branch

### PR DESCRIPTION
**Pull request by gsmendoza**
## Dependencies

This PR is currently on draft mode until https://github.com/solidusio/solidus_starter_frontend/pull/227 is merged.

## Motivation and Context

We want to test SolidusStarterFrontend against the latest Solidus master branch to ensure it will work with the next Solidus release.

## How Has This Been Tested?

Tested in CI.

## Types of changes
- N/A - Bug fix (non-breaking change which fixes an issue)
- N/A - New feature (non-breaking change which adds functionality)
- N/A - Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- N/A - My change requires a change to the documentation.
- N/A - I have updated the documentation accordingly.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
 
 commands:
+  setup:
+    steps:
+      - browser-tools/install-chrome
+      - browser-tools/install-chromedriver
+      - checkout
+      - run:
+          name: Install libvips
+          command: sudo apt-get install -y libvips
```
Comment (gsmendoza) [inline .circleci/config.yml:20]: libvips is required by Rails 7.

File: .circleci/config.yml [hunk 2]
```diff
@@ -91,6 +126,8 @@ workflows:
     jobs:
       - run-specs-with-postgres
       - run-specs-with-mysql
+      - run-specs-with-postgres-and-solidus-master
+      - run-specs-with-mysql-and-solidus-master
```
Comment (gsmendoza) [inline .circleci/config.yml:130]: We now have two sets of jobs: the first two jobs are for the current Solidus version and the next two jobs are for Solidus master.

File: Gemfile [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 source 'https://rubygems.org'
 
-gem 'rails'
+gem 'rails', ENV['RAILS_VERSION'] || '~> 7.0'
```
Comment (gsmendoza) [inline Gemfile:5]: Locally, the default Rails version is 7.0 while the default Solidus version is master.

File: templates/spec/controllers/spree/base_controller_spec.rb [hunk 4]
```diff
@@ -23,11 +23,14 @@ def index; authorize!(:read, :something); end
       end
 
       context "when http_referrer is present" do
-        before { request.env['HTTP_REFERER'] = '/redirect' }
+        let(:request_referer_path) { '/redirect' }
+        let(:request_referer) { "#{request.protocol}#{request.host}#{request_referer_path}" }
```
Comment (gsmendoza) [inline templates/spec/controllers/spree/base_controller_spec.rb:27]: We've updated the HTTP referers in the specs to match the test host. Rails 7.0 has a new default, wherein it protects against open redirects. Please see https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_to for more info.

File: .circleci/config.yml (hunk 2 above)
Comment (kennyadsl) [reply .circleci/config.yml:130]: Maybe we can rename the other two jobs into `run-specs-with-postgres-and-solidus-latest` to be more explicit?

**Review by kennyadsl — APPROVED** [review]
Left a non-blocking comment, feel free to merge anytime.

**Pull request merged**